tasks/upload.py

```diff
@@ -555,6 +556,27 @@ def schedule_task(
     ):
         commit_yaml = commit_yaml.to_dict()
 
+        # Carryforward the parent BA report for the current commit's BA report when handling uploads
```

Comment:
I think this block should be moved to a dedicated function function

Reply:
Done.